# Disable encryption of cookies, passwords, settings... on Mac.

--- a/components/os_crypt/os_crypt_mac.mm
+++ b/components/os_crypt/os_crypt_mac.mm
@@ -179,6 +179,10 @@ bool IsEncryptionAvailable() {
     return true;
   }
 
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch("disable-encryption")) {
+    return true;
+  }
+
   crypto::SymmetricKey* encryption_key = GetEncryptionKey();
   if (!encryption_key)
     return false;
@@ -218,6 +222,9 @@ bool IsEncryptionAvailable() {
       ciphertext.substr(strlen(kEncryptionVersionPrefix));
 
   crypto::SymmetricKey* encryption_key = GetEncryptionKey();
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch("disable-encryption")) {
+    return true;
+  }
   if (!encryption_key) {
     VLOG(1) << "Decryption failed: could not get the key";
     return false;
