Index: src/tools/mac/dump-static-initializers.py
===================================================================
--- src.orig/tools/mac/dump-static-initializers.py
+++ src/tools/mac/dump-static-initializers.py
@@ -41,7 +41,7 @@ def ParseDsymutil(binary):
   static initializers.
   """
 
-  child = subprocess.Popen(['tools/clang/dsymutil/bin/dsymutil', '-s', binary],
+  child = subprocess.Popen(['/usr/bin/dsymutil', '-s', binary],
      stdout=subprocess.PIPE)
   for line in child.stdout:
     file_match = dsymutil_file_re.search(line)
