# Removes the -Wno-ignored-pragma-optimize flag

--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1515,10 +1515,6 @@ config("default_warnings") {
           # instead of having a global flag.
           "-Wno-psabi",
 
-          # Ignore warnings about MSVC optimization pragmas.
-          # TODO(thakis): Only for no_chromium_code? http://crbug.com/912662
-          "-Wno-ignored-pragma-optimize",
-
           # TODO(https://crbug.com/989932): Evaluate and possibly enable.
           "-Wno-implicit-int-float-conversion",
 
@@ -1531,9 +1527,6 @@ config("default_warnings") {
           # TODO(https://crbug.com/1028110): Evaluate and possible enable.
           "-Wno-deprecated-copy",
 
-          # TODO(https://crbug.com/1050281): Clean up, enable.
-          "-Wno-non-c-typedef-for-linkage",
-
           # TODO(https://crbug.com/1114873): Clean up, enable.
           "-Wno-string-concatenation",
         ]
