description: crashpad relies on multicharacter integer assignments
author: Michael Gilbert <mgilbert@debian.org>

Index: src/third_party/crashpad/crashpad/BUILD.gn
===================================================================
--- src.orig/third_party/crashpad/crashpad/BUILD.gn
+++ src/third_party/crashpad/crashpad/BUILD.gn
@@ -17,6 +17,7 @@ import("build/test.gni")
 import("util/net/tls.gni")
 
 config("crashpad_config") {
+  cflags = [ "-Wno-multichar" ]
   include_dirs = [ "." ]
 }
 
