Index: src/third_party/node/node.py
===================================================================
--- src.orig/third_party/node/node.py
+++ src/third_party/node/node.py
@@ -14,6 +14,7 @@ def GetBinaryPath():
   # TODO: Node 16.0 will likely ship with an official universal node binary
   # on macOS. Once node 16.0 is released, remove this special case here
   # and use node-darwin-universal in the dict in the main return statement.
+  return subprocess.check_output(["which", "node"]).strip()
   if platform.system() == 'Darwin' and platform.machine() == 'arm64':
       return os.path.join(os_path.join(os_path.dirname(__file__), 'mac',
                           'node-darwin-arm64', 'bin', 'node'))
